File: C++/NIghtshadeBerries.cpp
#include "CONSTANTS.h"
#include "NightshadeBerries.h"

#include "World.h"

NightshadeBerries::NightshadeBerries(World &world, const Point &position) : Plant(world, position, NIGHTSHADE_BERRIES_SYMBOL, NIGHTSHADE_BERRIES_STRENGTH, NIGHTSHADE_BERRIES_COLOR, NIGHTSHADE_BERRIES_NAME) {}

bool NightshadeBerries::Collision(Organism *attacker) {
  this->world.messageManager.AddPlantEatingMessage(this->GetName(), attacker->GetName());
  this->Die();
  attacker->Collision(this);
  return false;
}

Organism* NightshadeBerries::Clone(const Point &position) {
  return new NightshadeBerries(this->world, position);
}

File: Java/NightshadeBerries.java
public class NightshadeBerries extends Plant {
  public NightshadeBerries(World world, Point position) {
    super(world, position, Constants.NIGHTSHADE_BERRIES_SYMBOL, Constants.NIGHTSHADE_BERRIES_STRENGTH, Constants.NIGHTSHADE_BERRIES_COLOR, Constants.NIGHTSHADE_BERRIES_NAME);
  }

  @Override
  public final Organism Clone(Point position) {
    return new NightshadeBerries(this.world, position);
  }

  @Override
  public final boolean Collision(Organism other) {
    this.world.messageManager.AddPlantEatingMessage(other.GetName(), this.GetName());
    this.world.messageManager.AddDeathMessage(other.GetName(), this.GetName());
    this.Die();
    other.Die();
    return false;
  }
}
